core/src/main/java/org/apache/brooklyn/core/catalog/internal/CatalogBomScanner.java

```diff
@@ -268,7 +268,7 @@ private void addLibraryDetails(Bundle bundle, Map<String, Object> catalog) {
 
         private String readBom(URL bom) {
             try (final InputStream ins = bom.openStream()) {
-                return Streams.readFullyString(ins);
+                return Streams.readFullyStringAndClose(ins);
```

Comment:
Haven't noticed that, will revert.
Yes, my thinking was that it doesn't really make sense to keep the stream open after reading it fully. I've updated the javadoc to encourage the usage of xxxAndClose alternatives. Only very specific cases will need to keep it unclosed.